### ISSUE TITLE
Spike: UPE PoC/Feasibility for classic checkout

### ISSUE THREAD

**Issue by c-shultz** [Automattic/woocommerce-payments#1625]
We wish to understand the capabilities of UPE and the scope of work (including plugin and server side) to integrate it for credit card and other payment method processing.

If possible, the goal of this spike will be to create a proof of concept to best discover feasibility/scope of this work by the end of the month.

**Comment by c-shultz** [comment]
@FangedParakeet ,

Please plan on P2'ing the results of this on or before Monday, May 3rd. We'll want to document any technical challenges/considerations to integrating UPE, and highlight any differences in UI flow/functionality between UPE and the standalone checkout elements.

**Comment by FangedParakeet** [comment]
@c-shultz definitely: I'm going to try to put together one or two quick PoC's over the next two days to try to illustrate a few different approaches and will document the hurdles and compromises I contend with along the way.

**Comment by FangedParakeet** [comment]
Just want to add the [link to the P2](paJDYF-1wJ-p2) I made containing the results of my investigation here. I've got a draft PR for a slightly janky, skeleton implementation here at #1714 .

I included a brief discussion on the impact on the Blocks checkout as well in the above P2 post, but to sum up, I don't believe there will be any blockers on that side with regards to implementing the UPE. I don't really see any blockers with moving forward with UPE at all at this stage. 

Storing saved payments [may require a little feedback from the Stripe team](paJDYF-1wJ-p2#comment-5813), as they haven't implemented this yet. In the meantime, it appears we can use saved payment methods the way they are currently implemented, presumably with a regular checkout. Recurring subscription payment methods may require specific implementations for specific payment methods: Sofort, for example, requires additional fields to be saved to process a recurring payment; however, it's unclear how this will be implemented within the UPE, as they are yet to implement this payment method at all. 

We will also need to add an endpoint to update a Payment Intention on the Payments Server.

These are things to be aware of moving forward, but I see no indication at this point that any of our ambitions will be left unsatisfied. There aren't really any obvious unsolved mysteries that I think need to be investigated right now, so I'm going to holster my guns for now, unless anyone else cries out another call for justice.
